addons/survey/controllers/main.py

```diff
@@ -670,17 +684,7 @@ def _extract_filters_data(self, survey, post):
         if line_choices:
             line_filter_domain = expression.AND([[('suggested_answer_id', 'in', line_choices)], line_filter_domain])
 
-        user_input_domain = ['&', ('test_entry', '=', False), ('survey_id', '=', survey.id)]
-        if line_filter_domain:
-            matching_line_ids = request.env['survey.user_input.line'].sudo().search(line_filter_domain).ids
-            user_input_domain = expression.AND([
-                [('user_input_line_ids', 'in', matching_line_ids)],
-                user_input_domain
-            ])
-        if post.get('finished'):
-            user_input_domain = expression.AND([[('state', '=', 'done')], user_input_domain])
-        else:
-            user_input_domain = expression.AND([[('state', '!=', 'new')], user_input_domain])
+        user_input_domain = self._get_user_input_domain(survey, line_filter_domain, post)
```

Comment:
Wondering if it would be cleaner to introduce a '_get_extra_user_input_domain' that would return False in survey.
So you don't split the domain building in multiple methods, which is a bit hard to follow.

So here you would have:
```py
extra_user_input_domain = self._get_extra_user_input_domain()
if extra_user_input_domain:
    user_input_domain = expression.AND(extra_user_input_domain, user_input_domain)

...

def _get_extra_user_input_domain(self, **post):
    return False
```

And in your appraisal you would simply add the necessary domain.
That simplifies parameters as well, you don't have to pass extra stuff (that you don't use at all in appraisal).

What do you think?